demos/imagenet_classification_demo/main.cpp

```diff
@@ -0,0 +1,503 @@
+                          fromHeight,
+                          modelInputResolution,
+                          std::min(modelInputResolution, resizedImage.rows - fromHeight));
+    }
+    resizedImage(imgROI).copyTo(resizedImage);
+
+    return resizedImage;
+}
+
+std::vector<std::vector<unsigned>> topResults(Blob& inputBlob, unsigned numTop) {
+    using currentBlobType = PrecisionTrait<Precision::FP32>::value_type;
+    TBlob<currentBlobType>& tblob = dynamic_cast<TBlob<currentBlobType>&>(inputBlob);
+    size_t batchSize =  tblob.getTensorDesc().getDims()[0];
+    numTop = static_cast<unsigned>(std::min<size_t>(size_t(numTop), tblob.size()));
+    
+    std::vector<std::vector<unsigned>> output(batchSize);
+    for (size_t i = 0; i < batchSize; i++) {
+        size_t offset = i * (tblob.size() / batchSize);
+        currentBlobType *batchData = tblob.data();
+        batchData += offset;
+        std::vector<unsigned> indices(tblob.size() / batchSize);
+        std::iota(std::begin(indices), std::end(indices), 0);
+        std::partial_sort(std::begin(indices), std::begin(indices) + numTop, std::end(indices),
+                          [&batchData](unsigned l, unsigned r) {
+                              return batchData[l] > batchData[r];
+                          });
+
+        indices.resize(numTop);
+        output[i] = std::move(indices);
+    }
+    
+    return output;
+}
+
+int main(int argc, char *argv[]) {
+    try {
+        if (!ParseAndCheckCommandLine(argc, argv)) {
+            return 0;
+        }
+
+        // -----------------------------------------Read input images-----------------------------------------
+        std::vector<std::string> imageNames;
+        std::vector<cv::Mat> inputImages;
+        parseInputFilesArguments(imageNames);
+        size_t inputImagesCount = imageNames.size();
+        if (inputImagesCount == 0) throw std::runtime_error("No images provided");
+        std::sort(imageNames.begin(), imageNames.end());
+        for (size_t i = 0; i < inputImagesCount; i++) {
+            const std::string& name = imageNames[i];
+            const cv::Mat& tmpImage = cv::imread(name);
+            if (tmpImage.data == nullptr) {
+                std::cerr << "Could not read image " << name << '\n';
+                imageNames.erase(imageNames.begin() + i);
+                inputImagesCount--;
+                i--;
+            } else {
+                inputImages.push_back(tmpImage);
+                imageNames[i] = name.substr(name.rfind('/') + 1);
+            }
+        }
+        // ---------------------------------------------------------------------------------------------------
+
+        // ----------------------------------------Read image classes-----------------------------------------
+        std::vector<unsigned> classIndices;
+        if (!FLAGS_gt.empty()) {
+            std::map<std::string, std::string> classIndicesMap;
+            std::ifstream inputClassesFile(FLAGS_gt);
```

Comment:
It should be checked that the file is opened. The same is for `inputLabelsFile`. The demo doesn't work if the file can't be opened.